### PR TITLE
Milestone 9: Finished Modal popup

### PR DESCRIPTION
**Pull request by prg-04**
## I have done the following to reach this milestone:

- [ ] Stored all the cards data dynamically
- [ ] Render the popup dynamically

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -13,6 +13,275 @@ toggle.addEventListener('click', () => {
...
+    divCls: ['popimg', 'poptext', 'popHead', 'topBtns', 'btns'],
+    classes: [
+      'fa-solid fa-x close',
+      'imgMob',
+      'imgDesk',
+      'popTtl',
+      'btn seeLive',
+      'btn seeLink',
+      'popLinks',
+      'link',
```
Comment (bobb-Rob) [inline main.js]: - [ ] I must commend your tenacity to push this project to this point. Well done 👏🏾. However, I see that you popup is not properly align in the center on the desktop view. Also, it is not scrolling down to make visible the text hidden. See image below to see your work in camparison to the Figma template.

Your work
image

Figma Template
image

You should try as much as possible to duplicate the Figma Template.

File: main.js [hunk 2]
```diff
@@ -13,6 +13,275 @@ toggle.addEventListener('click', () => {
...
+      'Codekit',
+      'Github',
+      'Javascript',
+      'Bootstrap',
+      'Terminal',
+      'Codepen',
+    ],
+    para: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry, Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the release",
+  },
+];
```
Comment (bobb-Rob) [inline main.js]: - [ ] As per the project requirement, you are expected to use an array to store a javascript object where each object contains at least the following information about the project: name, description, featured image, technologies, link to live version, and link to the project source code. Ie:

```javascript
const projects = [{
    id: 1
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
},
{
    id: 2
    name: "Helicopter game",
    description: "A Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
},
{
    id: 3
    name: "Cars racing game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]

// Then you will map through these projects array and display each project. 
```

See requirement 👇🏾 
image

File: main.js [hunk 3]
```diff
@@ -13,6 +13,275 @@ toggle.addEventListener('click', () => {
...
+  btn.innerText = txt;
+  return btn;
+}
+
+const work = document.getElementById('portfolio');
+
+const title = createDiv(cardsData[0].class);
+const h2 = createH2(cardsData[0].heading);
+title.appendChild(h2);
+work.appendChild(title);
```
Comment (bobb-Rob) [inline main.js]: - [ ] The names you give your variables matter a lot as they can help your coding partner quickly understand what your code is doing. The variable `title` could have been qualified as what kind of title, e.g `const workSectionTitle = .....`

Across your codebase, I see a lot of code that cannot easily make me understand what you are trying to achieve. I believe you can refactor your code and keep it simple. As a professional software engineer, you will work with many teams in your career, and writing easy-to-understand code will make you an effective team member. You may need to go over the [javascript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

File: main.js [hunk 4]
```diff
@@ -13,6 +13,275 @@ toggle.addEventListener('click', () => {
...
+const popup = document.querySelector('.popup');
+const collapse = document.querySelector('.fa-x');
+
+cardBtn.forEach((item) => {
+  item.addEventListener('click', () => {
+    popup.classList.add('open');
+    body.classList.add('blur');
+    hero.classList.add('blur');
+  });
+});
```
Comment (bobb-Rob) [inline main.js:263]: - [ ] Your popup menu currently is not dynamically displaying the correct project clicked. I will ask that you will implement this. When a user clicks on a `see project` button, that particular project information is displayed in the popup dynamically. eg. 

```javascript
// Grab all button
const allSeeProjectBtn = document.querySelectorAll(".trans-btn");
// Attach an onClick event on each button like below:
//Before you have attached the project Id to each buttons or any other means to get the id of the project clicked
allSeeProjectBtn.forEach((button) => {
  button.addEventListener("click", (e) => {
    const projectId = e.target.id;    
    // find the project in the projects array usin the id.
    const currentProject = projects.find((project) => project.id === projectId);
    // Make the pop up appear
    displayPopup(currentProject); //You will need to create a function that handles displaying popup.
    // grab the close button for this popup using DOM manipulation connections such as node.previousElementSibling, etc  
  });
});

```

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @prg-04 , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @prg-04,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Correct workflow was used ✔️
- No linter error ✔️
- Amazing styling ✔️
209362895-9762f2f8-13cf-4ce0-90b6-a672c4f04196

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**